File: smooth_player_api/model/songModel.js
const mongoose = require("mongoose")

const songSchema = new mongoose.Schema({
    title: {
        type: String,
        trim:true,
    },
    genre: {
        type: String,
    },
    album:{
        type: mongoose.Schema.ObjectId,
        ref: "album",
    },
    music_file:{
        type:String
    },
    cover_image:{
        type:String,
    },
    like:{
        type: Number,
        default:0,
    }
},
{
    timestamps: true
})

module.exports = mongoose.model("song", songSchema);

File: smooth_player_app/lib/api/model/song_model.dart
// ignore_for_file: non_constant_identifier_names

import 'dart:io';

class SongUploadModal {
  String? id;
  String? title;
  String? genre;
  File? music_file;
  File? cover_image;
  SongUploadModal({
    this.id,
    this.title,
    this.genre,
    this.music_file,
    this.cover_image,
  });
}
